stream_writer.go

```diff
@@ -74,18 +74,51 @@ func (db *DB) NewStreamWriter() *StreamWriter {
 // Prepare should be called before writing any entry to StreamWriter. It deletes all data present in
 // existing DB, stops compactions and any writes being done by other means. Be very careful when
 // calling Prepare, because it could result in permanent data loss. Not calling Prepare would result
-// in a corrupt Badger instance.
-func (sw *StreamWriter) Prepare() error {
+// in a corrupt Badger instance. It also supports an incremental stream write. In that case, it
+// writes the tables at one level above the current base level.
+func (sw *StreamWriter) Prepare(isIncremental bool) error {
```

Comment:
Can we rather expose an API `PrepareIncremental()`. It will make this a non-breaking change.